namespace-registry/registry.go

```diff
@@ -0,0 +1,411 @@
+		payload := url.Values{}
+		payload.Set("client_id", OIDCClientID)
+		payload.Set("client_secret", OIDCClientSecret)
+		payload.Set("device_code", device_code.(string))
+		payload.Set("grant_type", GrantType)
+
+		response, err := http.PostForm(TokenEndpoint, payload)
+		if err != nil {
+			log.Fatalln(err)
+		}
+		defer response.Body.Close()
+
+		body, err := ioutil.ReadAll(response.Body)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+		var tokenResponse TokenResponse
+		err = json.Unmarshal(body, &tokenResponse)
+		if err != nil {
+			log.Fatalln(err)
+		}
+
+
+		if tokenResponse.AccessToken == "" {
+			c.JSON(http.StatusOK, gin.H{
+				"status": "PENDING",
+			})
+		} else {
+			c.JSON(http.StatusOK, gin.H{
+				"status": "APPROVED",
+				"access_token": tokenResponse.AccessToken,
+			})
+		}
+		return 
+	}
+}
+
+func dbAddNamespace(c *gin.Context, data map[string]interface{}) {
+	var ns Namespace
+
+	ns.Prefix = data["prefix"].(string)
+	pubkeyData, _ := json.Marshal(data["pubkey"].(map[string]interface{}))
+	ns.Pubkey = string(pubkeyData)
+	if data["identity"] != nil {
+		ns.Identity = data["identity"].(string)
+	}
+
+	err := addNamespace(&ns)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"status": "success"})
+}
+
+func dbDeleteNamespace(c *gin.Context) {
+	prefix := c.Param("wildcard")
+
+	// A weird feature of gin is that wildcards always
+	// add a preceding /. We need to trim it here...
+	prefix = strings.TrimPrefix(prefix, "/")
+	log.Debug("Attempting to delete prefix", prefix)
+
+	err := deleteNamespace(prefix)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, gin.H{"status": "success"})
+}
+
+/**
+ * Commenting out until we're ready to use it.  -BB
+func cliListNamespaces(c *gin.Context) {
+	prefix := c.Param("prefix")
+	log.Debugf("Trying to get namespace data for prefix %s", prefix)
+	ns, err := getNamespace(prefix)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, ns)
+}
+*/
+
+func dbGetAllNamespaces(c *gin.Context) {
+	nss, err := getAllNamespaces()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+	c.JSON(http.StatusOK, nss)
+}
+
+/**
+ * Commenting out until we're ready to use it.  -BB
+func getJwks(c *gin.Context) {
+	prefix := c.Param("prefix")
+	c.JSON(http.StatusOK, gin.H{"status": "Get JWKS is not implemented", "prefix": prefix})
+}
+
+func getOpenIDConfiguration(c *gin.Context) {
+	prefix := c.Param("prefix")
+	c.JSON(http.StatusOK, gin.H{"status": "getOpenIDConfiguration is not implemented", "prefix": prefix})
+}
+*/
+
+func RegisterNamespaceRegistry(router *gin.RouterGroup) {
+	// Establish various routes to be used by the namespace registry
+	router.POST("/cli-namespaces/registry", cliRegisterNamespace)
```

Comment:
Let's dump the `cli-` prefix.  Affix this to the common API prefix used by other components.